module/surface_sam_sid_vhf.c

```diff
@@ -349,16 +349,16 @@ static int surface_sam_sid_vhf_probe(struct platform_device *pdev)
 		goto err_probe_hid;
 	}
 
-	status = hid_add_device(hid);
-	if (status) {
-		goto err_add_hid;
-	}
-
 	drvdata->event_ctx.dev = &pdev->dev;
 	drvdata->event_ctx.hid = hid;
 
 	platform_set_drvdata(pdev, drvdata);
 
+	status = hid_add_device(hid);
```

Comment:
@qzed Likely we want to do the `hid_add_device` even later, right before returning?

`hid-multitouch` will start doing things as soon as the HID device is detected

Reply:
I think this should (mostly) work as is. The core driver is able to send output reports and set/get feature reports without the events on the EC being disabled. It breaks when the driver expects some input report being sent from the EC as response (as that's not enabled at this point). I don't have any idea if that's possible or done in practice, so we might want to change that.

If we want to change that: I think we should first set up the EC events and enable them, and then register the HID device to ensure other drivers can actually use it before we add it. Similarly on removing, first de-register/destroy the HID device and then remove and disable the events. That will require us to check if the HID device is set up in the event handler function, Which should be possible via an atomic bool or something.